packages/augur-core/source/contracts/BaseSimpleDex.sol

```diff
@@ -0,0 +1,182 @@
+
+    event Mint(address indexed sender, uint256 tokenAmount, uint256 cashAmount);
+    event Burn(address indexed sender, uint256 tokenAmount, uint256 cashAmount, address indexed to);
+    event Swap(address indexed sender, address indexed tokenIn, uint256 amountIn, uint256 amountOut, address indexed to);
+    event Sync(uint256 tokenReserve, uint256 cashReserve);
+
+    function transferToken(address _to, uint256 _value) private;
+
+    function getTokenBalance() public returns (uint256);
+
+    function transferCash(address _to, uint _value) private {
+        cashTransfer(_to, _value);
+    }
+
+    function getCashBalance() public view returns (uint256) {
+        return cash.balanceOf(address(this));
+    }
+
+    function initializeInternal(address _augurAddress, address _token) internal {
+        endInitialization();
+        IAugur _augur = IAugur(_augurAddress);
+        augur = _augur;
+        cash = IERC20(_augur.lookup("Cash"));
+        require(cash != IERC20(0));
+        token = _token;
+
+        initializeCashSender(_augur.lookup("DaiVat"), address(cash));
+    }
+
+    function update(uint256 _tokenBalance, uint256 _cashBalance) private {
+        uint256 _blockNumber = block.number;
+        uint256 _blocksElapsed = _blockNumber - blockNumberLast;
+        uint256 _cashReserve = cashReserve;
+        uint256 _tokenReserve = tokenReserve;
+        if (_tokenReserve != 0 && _cashReserve != 0 && _blocksElapsed > 0) {
+            // cannot reasonably overflow unless the supply of Cash, REP, or ETH became several OOM larger
+            tokenPriceCumulativeLast += _cashReserve.mul(10**18).mul(_blocksElapsed).div(_tokenReserve);
+        }
+        tokenReserve = _tokenBalance;
+        cashReserve = _cashBalance;
+        blockNumberLast = _blockNumber;
+        emit Sync(_tokenBalance, _cashBalance);
+    }
+
+    function publicMint(address _to) external nonReentrant returns (uint256 _liquidity) {
+        uint256 _tokenBalance = getTokenBalance();
+        uint256 _cashBalance = getCashBalance();
+        uint256 _tokenReserve = tokenReserve;
+        uint256 _cashReserve = cashReserve;
+        uint256 _tokenAmount = _tokenBalance.sub(_tokenReserve);
+        uint256 _cashAmount = _cashBalance.sub(_cashReserve);
+
+        _liquidity = totalSupply == 0 ?
+            _tokenAmount.mul(_cashAmount).sqrt() :
+            (_tokenAmount.mul(totalSupply) / _tokenReserve).min(_cashAmount.mul(totalSupply) / _cashReserve);
+        require(_liquidity > 0, "Insufficient liquidity");
+        mint(_to, _liquidity);
+
+        update(_tokenBalance, _cashBalance);
+        emit Mint(msg.sender, _tokenAmount, _cashAmount);
+    }
+
+    function publicBurn(address _to) external nonReentrant returns (uint256 _tokenAmount, uint256 _cashAmount) {
+        uint256 _liquidity = balances[address(this)];
+
+        _tokenAmount = _liquidity.mul(tokenReserve) / totalSupply;
+        _cashAmount = _liquidity.mul(cashReserve) / totalSupply;
```

Comment:
Same comment as above, consider swapping the order of the variables to make it more clear that you are multiplying the reserve by the user's share of total supply.